src/main/java/ru/yandex/practicum/filmorate/service/user/UserService.java

```diff
@@ -99,8 +112,11 @@ public Collection<User> getCommonFriends(int id, int otherId) {
             ArrayList<User> commonUsers = new ArrayList<>();
 
             for (Integer userId : intersection) {
-                User u = userStorage.getUser(userId);
-                commonUsers.add(u);
+                Optional<User> optionalFriend = userStorage.getUser(userId);
+                if (optionalFriend.isPresent()) {
+                    User u = optionalFriend.get();
+                    commonUsers.add(u);
+                }
```

Comment:
можно попробовать использовать метод ifPresent у класса Optional, это немного сократит запись,
но тут еще есть момент, что мы в цикле обращаемся к базе, что не очень хорошо (на больших запросах долгое время ожидания, нагрузка на cpu)
можно сразу передавать список id, и в sql запросе оперировать списком. 
получится, что достанем данные одним запросом
```suggestion
                userStorage.getUser(userId).ifPresent(commonUsers::add);
```